File: config/src/main/java/com/typesafe/config/impl/ConfigNodeComment.java
package com.typesafe.config.impl;

import com.typesafe.config.ConfigException;
import com.typesafe.config.parser.ConfigNodeVisitor;

final class ConfigNodeComment extends ConfigNodeSingleToken implements com.typesafe.config.parser.ConfigNodeComment {
    ConfigNodeComment(Token comment) {
        super(comment);
        if (!Tokens.isComment(super.token)) {
            throw new ConfigException.BugOrBroken("Tried to create a ConfigNodeComment from a non-comment token");
        }
    }

    protected String commentText() {
        return Tokens.getCommentText(super.token);
    }

    @Override
    public <T> T accept(ConfigNodeVisitor<T> visitor) {
        return visitor.visitComment(this);
    }

    @Override
    public String getValue() {
        return commentText();
    }
}

File: config/src/main/scala/com/typesafe/config/impl/ConfigNodeComment.scala
package com.typesafe.config.impl

import com.typesafe.config.ConfigException

final class ConfigNodeComment(val comment: Token)
    extends ConfigNodeSingleToken(comment) {

    if (!Tokens.isComment(token))
        throw new ConfigException.BugOrBroken(
            "Tried to create a ConfigNodeComment from a non-comment token")

    private[impl] def commentText: String = Tokens.getCommentText(token)
}
